src/schemapack/_internals/validation/plugins/content_schema.py

```diff
@@ -71,8 +72,9 @@ def validate(
         Raises:
             schemapack.exceptions.ValidationPluginError: If validation fails.
         """
+        json_compatible_content = json.loads(resource.model_dump_json())["content"]
```

Comment:
Why is this necessary?

Reply:
The json standard library can only work with standard data types. Thus it does not know what FrozenDict is and you cannot make it aware of it. In deed it is doing isinstance checks directly on the commonly used standard types such as dict but not on generic types such as Mapping. The latter would have worked out of the box with FrozenDict. Thus we need to convert to standard dictionaries and the serialization option by pydantic is the easiest way to to so.